src/MaintenanceModeCommand.php

```diff
@@ -2,6 +2,154 @@
 
 namespace WP_CLI\MaintenanceMode;
 
-class MaintenanceModeCommand extends WP_CLI_Command {
+/**
+ * Activates, deactivates or checks the status of the maintenance mode of a site.
+ *
+ * ## EXAMPLES
+ *
+ *     # Activate Maintenance mode.
+ *     $ wp maintenance-mode activate
+ *     Enabling Maintenance mode...
+ *     Success: Activated Maintenance mode.
+ *
+ *     # Deactivate Maintenance mode.
+ *     $ wp maintenance-mode deactivate
+ *     Disabling Maintenance mode...
+ *     Success: Deactivated Maintenance mode.
+ *
+ *     # Display Maintenance mode status.
+ *     $ wp maintenance-mode status
+ *     Maintenance mode is active.
+ *
+ *     # Get Maintenance mode status for scripting purpose.
+ *     $ wp maintenance-mode is-active
+ *     $ echo $?
+ *     1
+ *
+ * @when after_wp_load
+ * @package wp-cli
+ */
+class MaintenanceModeCommand extends \WP_CLI_Command {
 
+
+	/**
+	 * Instance of the class.
+	 *
+	 * @var object Instance of the class.
+	 */
+	private static $instance;
+
+	/**
+	 * Return instance of the class.
+	 *
+	 * @return object
+	 */
+	public static function get_instance() {
```

Comment:
The entire flow with having a `get_instance()` call and storing the `WP_Upgrader` instance in a static variable is not clean OOP code, it is more procedural in nature.

Let's switch this to:
```
	/** @var \WP_Upgrader */
	private $upgrader;

	/**
	 * Instantiate a MaintenanceModeCommand object.
	 */
	public function __construct() {
		if ( ! class_exists( '\WP_Upgrader' ) ) {
			require_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
		}
		$this->upgrader = new \WP_Upgrader( new \WP_CLI\UpgraderSkin() );
		$this->upgrader->init();
	}
```

With this, you can just use the following in the other methods.:
```
$this->upgrader->maitenance_mode( true );
```